### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by subfuzion**
This is just a bit of refactoring/restructuring to make the things we still need to do a bit clearer and to iterate somewhat on the README content. The content under `_docker-library` is ultimately intended to be removed from this repo and become pull requests to two other repos:

- https://github.com/docker-library/docs
- https://github.com/docker-library/official-images

We could work directly with clones of those, but it might be easier to have all the content in this repo while we iterate on it.

Also note that I followed the pattern used by the official image for Node.js (which has `docker-library/docs/node/README.me` point to `nodejs/docker-node/README.md` for how to use the image, rather than the other way around like the Go team does; since we're following the Node model a bit more by maintaining our own official repo on dart-lang outside of the docker-library org, this probably makes more sense for us.

### CODE REVIEW

**Comment by mit-mit** [comment]
I suggest we change the README in a separate PR to keep the changes more isolated. Here's a small PR for that: https://github.com/dart-lang/dart-docker/pull/10

File: README.md [hunk 1]
```diff
@@ -3,14 +3,44 @@
 This is the Git repo of the [Docker "Official Images"] for the
 [Dart programming language].
 
-See the [Docker Hub page] for a full description on how to use these Docker
-images as well as for information regarding contributing and issues.
+## How to use this image
```
Comment (mit-mit) [inline README.md:6]: Sync with changes from https://github.com/dart-lang/dart-docker/pull/10

File: _docker-library/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# _docker-library
+
+This directory is a placeholder with samples for the following repos:
```
Comment (mit-mit) [inline _docker-library/README.md:3]: I'm a bit confused, I thought we talked about *not* having samples here?

Comment (subfuzion) [reply _docker-library/README.md:3]: These are not samples for how to use our repo. These are placeholder samples of what *we* need to submit to the other repos as part of the process of creating of publishing our official image.

File: README.md (hunk 1 above)
Comment (subfuzion) [reply README.md]: Done.

**Comment by kevmoo** [comment]
I don't have all of the context, but this generally seems like we're moving the ball forward.

**Pull request merged**